lib/LWP/UserAgent.pm

```diff
@@ -747,6 +747,23 @@ sub cookie_jar {
     $old;
 }
 
+sub proxy_header {
+    my $self = shift;
+    $self->{proxy_headers} ||= HTTP::Headers->new;
+    return $self->{proxy_headers}->header(@_);
+}
+
+sub proxy_headers {
+    my $self = shift;
+    my $old = $self->{proxy_headers} ||= HTTP::Headers->new;
+    if (@_) {
+	Carp::croak("proxy_headers not set to HTTP::Headers compatible object")
+	    unless @_ == 1 && $_[0]->can("header_field_names");
```

Comment:
I have use the same checks as `default_headers` uses. Usage of `proxy_headers` and `default_headers` should be the same, shouldn't it?